pkgs/tools/admin/balena-cli/default.nix

```
@@ -0,0 +1,68 @@
+{ lib
+, stdenv
+, fetchurl
+, unzip
+, installShellFiles
+}:
+
+let
+  inherit (stdenv.hostPlatform) system;
+  throwSystem = throw "Unsupported system: ${system}";
+
+  plat = {
+    x86_64-linux = "linux-x64";
+    x86_64-darwin = "macOS-x64";
+    
+    # Balena only packages for x86 so we rely on Rosetta for Apple Silicon
+    aarch64-darwin = "macOS-x64";
+    x86_64-windows = "windows-x64";
+  }.${system} or throwSystem;
+
+  sha256 = {
+    x86_64-linux = "1cf1047303885e1df8ccbb30afde91b75d84ddeaa69e880998f37be17553baa7";
+    x86_64-darwin = "692eb4a9940c8115d8c4c4c9ee97b9da7840d6bee6863b1ec5a05336c67dc9b9";
+    aarch64-darwin = "692eb4a9940c8115d8c4c4c9ee97b9da7840d6bee6863b1ec5a05336c67dc9b9";
+    x86_64-windows = "7ba3629f6d052d4947629da158482ec30ee6efe90fdf8ca3b79a8f6a04e4935a";
+  }.${system} or throwSystem;
+in
+stdenv.mkDerivation rec {
+  pname = "balena-cli";
+  version = "15.1.1";
+
+  src = fetchurl {
+    url = "https://github.com/balena-io/balena-cli/releases/download/v${version}/balena-cli-v${version}-${plat}-standalone.zip";
+    inherit sha256;
+  };
+
+  nativeBuildInputs = [ unzip ];
+
+  installPhase = ''
+    runHook preInstall
+
+    mkdir -p $out/bin
+    cp -r ./* $out/
+
+    ln -s $out/balena $out/bin/balena
+
+    runHook postInstall
+  '';
+
+  meta = with lib; {
+    mainProgram = "balena";
```

```suggestion
    mainProgram = "balena";
```
Please move mainProgram to bottom of meta. Having description at top is best. There is order to meta attributes. I know it may sound odd. But there are reasons for this. For example, `broken` is always last at meta so it doesn't create git conflicts when marking/unmarking broken.